src/test/components/UrlManager.test.js

```diff
@@ -104,14 +101,17 @@ describe('UrlManager', function() {
     expect(getDataSource(getState())).toMatch('none');
   });
 
-  it('sets the data source to public', async function() {
+  it(`sets the data source to public and doesn't change the URL when there's an error`, async function() {
+    const urlPath = '/public/FAKE_HASH/marker-chart';
     const { getState, createUrlManager, waitUntilUrlSetupPhase } = setup(
-      '/public/'
+      urlPath
     );
     expect(getDataSource(getState())).toMatch('none');
     createUrlManager();
 
     await waitUntilUrlSetupPhase('done');
     expect(getDataSource(getState())).toMatch('public');
+    expect(getView(getState()).phase).toBe('FATAL_ERROR');
+    expect(window.location.pathname).toBe(urlPath);
```

Comment:
I checked that this is failing as expected on master :-)

Reply:
Cool, thanks!